frontend/src/api/LogApi.js

```diff
@@ -20,10 +22,36 @@ async function getLogs(endpoint) {
     });
 }
 
+async function getLogsOfStream(endpoint) {
+  return axios
+    .get(endpoint, getAuthorizationHeader())
+    .then((response) => {
+      return Promise.resolve(
+        response.data.map((log) => {
+          // log.serialNumber = sender, log.logEntity.serialNumber = receiver
+          return new StreamLogInfo(
+            log.logEntity.dateTime,
+            log.logEntity.level,
+            log.serialNumber,
+            log.logEntity.serialNumber,
+            log.logEntity.message
+          );
+        })
+      );
+    })
+    .catch(() => {
+      snackbar("error", "Failed to fetch stream logs");
+    });
```

Comment:
Displaying the snackbar should not be a responsibility of the API level, this should be done in a `.catch` in the component that calls this function.

Also, it might be a good idea to include the response error code, if present, to aid with debugging.

Reply:
1b904e1 370581c